### PR TITLE
1.20.1 rtm alloy coils

### PR DESCRIPTION
**Pull request by Markh3tr**
Ported the RTM Alloy Coils from 1.12.2 GTCEu 2.8.

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/data/lang/MaterialLangGenerator.java [hunk 1]
```diff
@@ -2,15 +2,19 @@
...
+import static com.gregtechceu.gtceu.data.lang.LangHandler.replace;
 
 public class MaterialLangGenerator {
 
     public static void generate(RegistrateLangProvider provider, MaterialRegistry registry) {
         for (Material material : registry.getAllMaterials()) {
             provider.add(material.getUnlocalizedName(), toEnglishName(material.getName()));
         }
+
+        replace(provider, GTMaterials.RTMAlloy.getUnlocalizedName(),"RTM Alloy");
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/data/lang/MaterialLangGenerator.java]: Wrong class

File: src/main/java/com/gregtechceu/gtceu/common/block/CoilBlock.java [hunk 2]
```diff
@@ -63,7 +63,8 @@ public enum CoilType implements StringRepresentable, ICoilType {
         CUPRONICKEL("cupronickel", 1800, 1, 1, GTMaterials.Cupronickel, GTCEu.id("block/casings/coils/machine_coil_cupronickel")),
         KANTHAL("kanthal", 2700, 2, 1, GTMaterials.Kanthal, GTCEu.id("block/casings/coils/machine_coil_kanthal")),
         NICHROME("nichrome", 3600, 2, 2, GTMaterials.Nichrome, GTCEu.id("block/casings/coils/machine_coil_nichrome")),
-        TUNGSTENSTEEL("tungstensteel", 4500, 4, 2, GTMaterials.TungstenSteel, GTCEu.id("block/casings/coils/machine_coil_tungstensteel")),
+        RTMALLOY("rtm_alloy", 4500, 4, 2, GTMaterials.RTMAlloy, GTCEu.id("block/casings/coils/machine_coil_rtm_alloy")),
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/common/block/CoilBlock.java:66]: You do realize that this will delete everyone's tungstensteel coils right?

**Review by screret — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregtechceu/gtceu/common/block/CoilBlock.java (hunk 2 above)
Comment (Markh3tr) [reply src/main/java/com/gregtechceu/gtceu/common/block/CoilBlock.java:66]: I do not, how would I do this properly?

**Comment by screret** [comment]
[fix_tungstensteel_coils_disappearing_with_change_to_RTM_alloy.patch](https://github.com/GregTechCEu/GregTech-Modern/files/14229088/fix_tungstensteel_coils_disappearing_with_change_to_RTM_alloy.patch)
apply this patch it remaps the block id when loading the world

File: src/main/java/com/gregtechceu/gtceu/data/lang/MaterialLangGenerator.java (hunk 1 above)
Comment (Markh3tr) [reply src/main/java/com/gregtechceu/gtceu/data/lang/MaterialLangGenerator.java]: What class should I put this in?

File: src/generated/resources/assets/gtceu/lang/en_us.json [hunk 3]
```diff
@@ -765,6 +765,7 @@
   "block.gtceu.pyrolyse_oven": "Pyrolyse Oven",
   "block.gtceu.reaction_safe_mixing_casing": "Reaction-Safe Mixing Casing",
   "block.gtceu.robust_machine_casing": "Robust Tungstensteel Machine Casing",
+  "block.gtceu.rtm_alloy_coil_block": "Rtm Alloy Coil Block",
```
Comment (screret) [inline src/generated/resources/assets/gtceu/lang/en_us.json]: fix this lang & we're golden

**Review by screret — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregtechceu/gtceu/data/lang/MaterialLangGenerator.java (hunk 1 above)
Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/data/lang/MaterialLangGenerator.java]: LangHandler, with the HSS-G replacement etc.

Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/data/lang/MaterialLangGenerator.java]: because MaterialLangGenerator generates per-MOD, including addons

**Review by screret — APPROVED** [review]

**Pull request merged**